SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditor.cpp

```diff
@@ -97,18 +97,23 @@ FSpatialGDKEditor::FSpatialGDKEditor()
 {
 }
 
-bool FSpatialGDKEditor::GenerateSchema(ESchemaGenerationMethod Method)
+TFuture<bool> FSpatialGDKEditor::GenerateSchema(ESchemaGenerationMethod Method)
 {
+	TSharedPtr<TPromise<bool>> Result = MakeShared<TPromise<bool>>();
```

Comment:
A future is returned here so the task can stay open whilst the UAT job runs (completed in a callback)

Reply:
Question: Why do we have to make a promise and not just a future? In other words why is it not:

 MakeShared<TFuture<bool>>();

Reply:
There's no way to set the value on a future (the promise and future are tied together, the only way being setting the promise value to update the future)